### PR TITLE
[WIP] 1.13 Rendering patches

### PR DESCRIPTION
**Pull request by tterrag1098**
* net.minecraft.client.renderer:
  - [x] ActiveRenderInfo
  - [x] BlockFluidRenderer
  - [x] BlockModelRenderer
  - [x] BlockModelShapes (Skipped)
  - [x] BlockRendererDispatcher
  - [x] BufferBuilder (Skipped, all fixes implemented in vanilla)
  - [x] ChestRenderer
  - [x] EntityRenderer (new patches for LightTexture and FogRenderer)
  - [x] InventoryEffectRenderer
  - [x] ItemModelMesher 
  - [x] ItemRenderer (now `FirstPersonRenderer`, add new patch for ItemMap)
  - [x] OpenGlHelper
  - [x] RenderGlobal
  - [x] RenderItem
  - [x] WorldVertexBufferUploader
* net.minecraft.client.renderer.block.model:
  - [x] BakedQuad
  - [x] BakedQuadRetextured
  - [x] FaceBakery
  - [x] IBakedModel
  - [x] ItemCameraTransforms
  - [x] ItemOverrideList
  - [x] ItemTransformVec3f
  - [ ] ModelBakery
  - [x] ModelBlockDefinition
  - [x] ModelManager
  - [x] ModelResourceLocation
  - [x] ModelRotation
  - [x] SimpleBakedModel
  - [x] Variant
  - [x] WeightedBakedModel
* net.minecraft.client.renderer.block.statemap:
  - [x] StateMap (gone)
* net.minecraft.client.renderer.chunk:
  - [x] ChunkRenderDispatcher
  - [x] RenderChunk
* net.minecraft.client.renderer.color:
  - [x] BlockColors
  - [x] ItemColors
* net.minecraft.client.renderer.entity:
  - [x] RenderEntityItem
  - [x] RenderFish
  - [x] RenderItemFrame
  - [ ] RenderLivingBase
  - [x] RenderManager
  - [x] RenderPlayer
  - [x] RenderVillager
  - [x] RenderZombieVillager
* net.minecraft.client.renderer.entity.layers:
  - [x] LayerArmorBase
  - [x] LayerBipedArmor
  - [x] LayerHeldItem
* net.minecraft.client.renderer.texture:
  - [x] Stitcher
  - [x] TextureAtlasSprite
  - [x] TextureManager
  - [x] TextureMap (This is a huge mess, only a first pass done)
  - [x] TextureUtil
* net.minecraft.client.renderer.tileentity:
  - [x] TileEntityItemStackRenderer (Deprecated, removed)
  - [x] TileEntityPistonRenderer (No longer necessary (?))
  - [x] TileEntityRendererDispatcher
  - [x] TileEntitySpecialRenderer
* net.minecraft.client.renderer.vertex:
  - [x] VertexFormat
  - [x] VertexFormatElement

### CODE REVIEW

File: patches/minecraft/net/minecraft/client/renderer/LightTexture.java.patch [hunk 1]
```diff
@@ -0,0 +1,19 @@
+--- a/net/minecraft/client/renderer/LightTexture.java
++++ b/net/minecraft/client/renderer/LightTexture.java
+@@ -110,7 +110,16 @@
+                      f11 = 0.28F + f8 * 0.75F;
+                      f12 = 0.25F + f9 * 0.75F;
+                   }
++                  
++                  float[] colors = {f10, f11, f12};
++                  world.func_201675_m().getLightmapColors(p_205106_1_, f, f4, f5, colors);
++                  f10 = colors[0]; f11 = colors[1]; f10 = colors[2];
```
Comment (bs2609) [inline patches/minecraft/net/minecraft/client/renderer/LightTexture.java.patch]: Minor typo here - second `f10` should be `f12`

Comment (tterrag1098) [reply patches/minecraft/net/minecraft/client/renderer/LightTexture.java.patch]: Darn, and I thought I caught them all.

File: patches/minecraft/net/minecraft/client/renderer/block/model/IBakedModel.java.patch [hunk 2]
```diff
@@ -1,6 +1,18 @@
 --- a/net/minecraft/client/renderer/block/model/IBakedModel.java
 +++ b/net/minecraft/client/renderer/block/model/IBakedModel.java
-@@ -21,7 +21,18 @@
+@@ -8,9 +8,10 @@
+ import net.minecraft.util.EnumFacing;
+ import net.minecraftforge.api.distmarker.Dist;
+ import net.minecraftforge.api.distmarker.OnlyIn;
++import net.minecraftforge.client.extensions.IForgeBakedModel;
```
Comment (ichttt) [inline patches/minecraft/net/minecraft/client/renderer/block/model/IBakedModel.java.patch]: Import in patch

File: patches/minecraft/net/minecraft/client/renderer/chunk/ChunkRenderDispatcher.java.patch [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    private final ChunkRenderWorker field_178525_i;
+ 
+    public ChunkRenderDispatcher() {
++      this(-1);
++   }
++   
++   public ChunkRenderDispatcher(int countRenderBuilders) {
+       int i = Math.max(1, (int)((double)Runtime.getRuntime().maxMemory() * 0.3D) / 10485760);
+       int j = Math.max(1, MathHelper.func_76125_a(Runtime.getRuntime().availableProcessors(), 1, i / 5));
++      if(countRenderBuilders < 0) countRenderBuilders = MathHelper.func_76125_a(j * 10, 1, i);
```
Comment (ichttt) [inline patches/minecraft/net/minecraft/client/renderer/chunk/ChunkRenderDispatcher.java.patch:14]: This parameter seems like it is never read from

File: patches/minecraft/net/minecraft/client/renderer/entity/layers/LayerArmorBase.java.patch [hunk 4]
```diff
@@ -0,0 +1,98 @@
+--- a/net/minecraft/client/renderer/entity/layers/LayerArmorBase.java
++++ b/net/minecraft/client/renderer/entity/layers/LayerArmorBase.java
+@@ -2,6 +2,9 @@
+ 
+ import com.google.common.collect.Maps;
+ import java.util.Map;
++
++import javax.annotation.Nullable;
```
Comment (ichttt) [inline patches/minecraft/net/minecraft/client/renderer/entity/layers/LayerArmorBase.java.patch]: Import

File: patches/minecraft/net/minecraft/client/renderer/texture/TextureMap.java.patch [hunk 5]
```diff
@@ -0,0 +1,186 @@
+--- a/net/minecraft/client/renderer/texture/TextureMap.java
++++ b/net/minecraft/client/renderer/texture/TextureMap.java
+@@ -7,6 +7,7 @@
+ import java.io.IOException;
+ import java.util.List;
+ import java.util.Map;
++import java.util.OptionalInt;
```
Comment (ichttt) [inline patches/minecraft/net/minecraft/client/renderer/texture/TextureMap.java.patch]: Import

**Review by ichttt — COMMENTED** [review]
Just looked through the patches on github real quick, this is what I noticed. Not a complete review.

File: patches/minecraft/net/minecraft/client/renderer/chunk/ChunkRenderDispatcher.java.patch (hunk 3 above)
Comment (tterrag1098) [reply patches/minecraft/net/minecraft/client/renderer/chunk/ChunkRenderDispatcher.java.patch:14]: Intended, this is just a hook for mods

Comment (ichttt) [reply patches/minecraft/net/minecraft/client/renderer/chunk/ChunkRenderDispatcher.java.patch:14]: But it is not even written to a field in this patch, it just calculates something and then never accesses it. In 1.12 the result is written to the field.

Comment (tterrag1098) [reply patches/minecraft/net/minecraft/client/renderer/chunk/ChunkRenderDispatcher.java.patch:14]: You're right, good catch.

**Pull request merged**